src/fluree/db/store.cljc

```diff
@@ -0,0 +1,74 @@
+(ns fluree.db.store
+  (:refer-clojure :exclude [read list exists?])
+  (:require [fluree.db.store.proto :as store-proto]
+            [fluree.db.store.file :as file-store]
+            [fluree.db.store.localstorage :as localstorage-store]
+            [fluree.db.store.memory :as mem-store]
+            [malli.core :as m]))
+
+
+(def BaseConfig
+  [:map
+   [:store/method [:enum :memory :localstorage :file :ipfs :s3 :remote]]])
+
+(def FileConfig
+  [:and
+   BaseConfig
+   [:map
+    [:store/method [:enum :file]]
+    [:file-store/storage-path :string]]])
+
+(def LocalStorageConfig
+  [:and
+   BaseConfig
+   [:map
+    [:store/method [:enum :localstorage]]]])
+
+(def MemoryConfig
+  [:and
+   BaseConfig
+   [:map
+    [:store/method [:enum :memory]]
+    [:memory-store/storage-atom {:optional true} :any]]])
+
+(def StoreConfig
+  [:or
+   FileConfig
+   LocalStorageConfig
+   MemoryConfig])
+
+(defn start
+  [{:keys [:store/method] :as config}]
+  (if-let [config-error (m/explain StoreConfig config)]
+    (throw (ex-info "Invalid Store config."
+                    {:error  config-error
+                     :config config}))
+    (case method
+      :file         (file-store/create-file-store config)
+      :localstorage (localstorage-store/create-localstorage-store config)
+      :memory       (mem-store/create-memory-store config)
+
+      (throw (ex-info (str "No Store implementation for :store/method: " (pr-str method))
+                      config)))))
+
+(defn write
```

Comment:
As it stands now, these functions will be redundant when we move the protocol into this namespace. There might be more functionality we build into them later, but this redundancy is part of the reason the protocol should be here to begin with.